community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java

```diff
@@ -900,6 +912,18 @@ void open( EphemeralFileChannel channel )
             }
         }
 
+        void force()
+        {
+            forcedBuffer = fileAsBuffer.copy();
+            forcedSize = size;
+        }
+
+        void crash()
+        {
+            fileAsBuffer = forcedBuffer.copy();
+            size = forcedSize;
```

Comment:
The write method synchronises on the `fileAsBuffer` field and updates the size, so I think the locking in here might need an overhaul.